src/matchengine/controller.rs

```diff
@@ -219,24 +220,28 @@ impl Controller {
         } else {
             req.limit
         };
-        let market = self
+        let markets = self
             .markets
-            .get(&req.market)
-            .ok_or_else(|| Status::invalid_argument("invalid market"))?;
-        let total_order_count = market.users.get(&req.user_id).map(|order_map| order_map.len()).unwrap_or(0);
-        let orders = market
-            .users
-            .get(&req.user_id)
-            .map(|order_map| {
-                order_map
-                    .values()
-                    .rev()
-                    .skip(req.offset as usize)
-                    .take(limit as usize)
-                    .map(|order_rc| OrderInfo::from(order_rc.deep()))
-                    .collect()
+            .iter()
+            .filter(|(key, _market)| req.market == "all" || req.market == **key)
+            .map(|(_key, market)| market);
+        let total_order_count: usize = markets
+            .clone()
+            .map(|m| m.users.get(&req.user_id).map(|order_map| order_map.len()).unwrap_or(0))
+            .sum();
+        let orders_by_market: Vec<Box<dyn Iterator<Item = Order>>> = markets
+            .map(|m| {
+                m.users
+                    .get(&req.user_id)
+                    .map(|order_map| Box::new(order_map.values().rev().map(|order_rc| order_rc.deep())) as Box<dyn Iterator<Item = Order>>)
+                    .unwrap_or_else(|| Box::new(Vec::new().into_iter()) as Box<dyn Iterator<Item = Order>>)
             })
-            .unwrap_or_else(Vec::new);
+            .collect();
+        let orders = MergeSortIterator::compare_by(orders_by_market, SortOrder::Asc, |a, b| a.id.cmp(&b.id))
```

Comment:
compare by `create_time`?

Reply:
it was sort by id in original version.

Reply:
but in https://github.com/Fluidex/dingir-exchange/issues/231 @lispc says sort by `create_time`?

Reply:
revert commit  79f675d